internal/resto/client.go

```diff
@@ -429,3 +436,55 @@ func doRequestAsset(httpClient *http.Client, request *http.Request) error {
 	}
 	return nil
 }
+
+// DownloadArtifact does downloading artifacts
+func (c *Client) DownloadArtifact(jobID string) {
+	targetDir := filepath.Join(c.ArtifactConfig.Directory, jobID)
+	if err := os.MkdirAll(targetDir, 0755); err != nil {
+		log.Error().Msgf("Unable to create %s to fetch artifacts (%v)", targetDir, err)
+		return
+	}
+
+	files, err := c.GetJobAssetFileNames(context.Background(), jobID)
+	if err != nil {
+		log.Error().Msgf("Unable to fetch artifacts list (%v)", err)
+		return
+	}
+	for _, f := range files {
+		for _, pattern := range c.ArtifactConfig.Match {
+			if glob.Glob(pattern, f) {
+				if err := c.downloadArtifact(targetDir, jobID, f); err != nil {
+					log.Error().Err(err).Msgf("Failed to download file: %s", f)
+				}
+				break
+			}
+		}
+	}
+}
+
+func (c *Client) downloadArtifact(targetDir, jobID, fileName string) error {
+	content, err := c.GetJobAssetFileContent(context.Background(), jobID, fileName)
+	if err != nil {
+		return err
+	}
+	targetFile := filepath.Join(targetDir, fileName)
+	return os.WriteFile(targetFile, content, 0644)
+}
+
+// ShouldDownload returns true if it should download artifacts, otherwise false
+func ShouldDownload(jobID string, passed bool, cfg config.ArtifactDownload) bool {
```

Comment:
Looks all good aside from this part 😄  `resto` is an implementation detail. Whether or not an artifact is to be downloaded, doesn't depend on resto though. This logic is purely functional business logic and does not depend on resto, so it should be somewhere else 💡

Reply:
I put the function into `download` package 🤔